test/core_test.clj

```diff
@@ -1,51 +1,64 @@
 (ns core-test
   (:require [clojure.java.io :as io]
-            [clojure.test :refer [deftest is testing run-tests]]
+            [clojure.test :refer [deftest is run-tests testing]]
             [bb-excel.core :refer [get-sheets get-sheet-names get-sheet
                                    get-range
                                    ; get-row get-col get-cells crange
                                    ]])
-  (:import [java.util.zip ZipFile]))
+  (:import (clojure.lang ExceptionInfo)
+           [java.util.zip ZipFile]))
+
+(declare thrown-with-msg?) ;; Workaround from https://github.com/cursive-ide/cursive/issues/238
 
 (deftest zipfile-or-nil-test
-  (let [zipfile-or-nil #'bb-excel.core/zipfile-or-nil]
+  (let [zipfile-or-nil #'bb-excel.core/get-zipfile]
     (let [file (io/file "test/data/simple.xlsx")]
       (is (instance? ZipFile (zipfile-or-nil file))))
     (let [filepath "test/data/simple.xlsx"]
       (is (instance? ZipFile (zipfile-or-nil filepath))))
-    (is (nil? (zipfile-or-nil "invalid-file-path")))
-    (is (nil? (zipfile-or-nil :invalid-type)))))
+    (is (thrown-with-msg? ExceptionInfo #"Could not open 'invalid-file-path'! File does not exist."
+                          (zipfile-or-nil "invalid-file-path")))
+    (is (thrown-with-msg? ExceptionInfo #"Could not open ':invalid-type'! Argument should be string or file."
+                          (zipfile-or-nil :invalid-type)))))
```

Comment:
I not sure anyone wants return nil instead of error in this cases, but maybe you see it another way